airflow/settings.py

```diff
@@ -207,6 +209,35 @@ def configure_vars():
     DONOT_MODIFY_HANDLERS = conf.getboolean("logging", "donot_modify_handlers", fallback=False)
 
 
+def _run_openlineage_runtime_check():
+    """
+    Ensure compatibility of OpenLineage provider package and Airflow version.
+
+    Airflow 2.10.0 introduced some core changes (#39336) that made versions <= 1.8.0 of OpenLineage
+    provider incompatible with future Airflow versions (>= 2.10.0).
+    """
+    ol_package = "apache-airflow-providers-openlineage"
+    try:
+        ol_version = metadata.version(ol_package)
```

Comment:
I think we should have an extra check for breeze. In breeze, there is no provider installed - providers are read directly from sources, so we should rather check version of OpenLineage Provider via ProvidersManager, not directly via importlib

Reply:
Done. That's the first time I'm using it, so not sure if it's correct.

Reply:
**actually** I just realized that it will work just fine as well - because .... provider will not be installed in this case and we just return. :facepalm:

Reply:
When using `source` installation maybe we should just emit a warning? That's usually a development scenario anyway, right? The one that matters the most here is users installing the package.

Also then the CI would probably pass, now with 2.10.0.dev0 the error is raise as current provider is 1.7.1 (the try_num adjustment has not been released yet, it's only in source)

Reply:
Yeah. Let's just use importlib. Will be simpler.

Reply:
Not sure i understand it tho. I mean, i used the importlib here now, but isn't that what the `ProvidersManager` uses underneath when looking for packages installed (airflow.utils.entry_points._get_grouped_entry_points)? I thought that when `ProvidersManager().providers.get(ol_package).is_source` is False then `ProvidersManager().providers.get(ol_package).version` , it's the same as `importlib.metadata.version(ol_package)` - it simply detects whether or not package has been installed.

So i could just write this, and it should work just fine, right?
```
    ol_package = "apache-airflow-providers-openlineage"
    ol_provider = ProvidersManager().providers.get(ol_package)
    if not ol_provider:
        return

    err_msg = (
        f"You have installed `{ol_package}` == `{ol_provider.version}` that is not compatible with "
        f"`apache-airflow` == `{airflow_version}`. "
        f"For `apache-airflow` >= `2.10.0` you must use `{ol_package}` >= `1.8.0`."
    )

    if ol_provider.version and Version(ol_provider.version) < Version("1.8.0"):
        if ol_provider.is_source:
            log.warning(err_msg)
        else:
            raise RuntimeError(err_msg)
```

Reply:
It could be done this way, yes. But also there is no need to use ProvidersManager at all. In this case if provider is installed from sources, it's not installed as package at all so when providers are installed from sources this if will return immediately (same as openlineage provider not installed). Which is what we expect - because when we use airflow from sources, we know both airflow and provider are correctly latest versions.

```
    ol_provider = importlib.verison("apache-airflow-providers-openlinege") 
    if not ol_provider:
        return
```

Reply:
Got it, it makes sense to skip the check when installing from source. Reverted the code to the original one, just using importlib.